Source/WebKit/UIProcess/VisitedLinkStore.cpp

```diff
@@ -133,7 +133,7 @@ void VisitedLinkStore::didUpdateSharedStringHashes(const Vector<WebCore::SharedS
     for (Ref process : m_processes) {
         ASSERT(process->processPool().processes().containsIf([&](auto& item) { return item.ptr() == process.ptr(); }));
 
-        if (addedHashes.size() > 20 || !removedHashes.isEmpty())
+        if (addedHashes.size() > 20 || !removedHashes.isEmpty() || process->throttler().isSuspended())
```

Comment:
should it be `!process->throttler().isSuspended()`?

Reply:
I think this is what I am going for. If the receiver is suspended, then hopefully it will stay suspended for a while, and we'll just ask it to reevaluate the state of all visited links when it wakes up (by sending AllVisitedLinkStateChanged).

If receiver is not suspended, and the number of link state changes is small, then continue sending the VisitedLinkStateChanged message. Those messages can't be coalesced with the current DeferSendingIfSuspended logic.

I could also do something better if I were to write custom logic here, but hopefully sending AllVisitedLinkStateChanged to suspended processes isn't too wasteful.

Reply:
No this makes sense, I missed the `All` prefix in the message name in this branch.